modules/tinymce/src/plugins/table/test/ts/module/test/TableTestUtils.ts

```diff
@@ -415,6 +444,53 @@ const sAssertTableStructureWithSizes = (editor: Editor, cols: number, rows: numb
   })
 ]);
 
+const sAssertTableStructureWithSizesAndColGroups = (
+  editor: Editor,
+  cols: number,
+  rows: number,
+  unit: string | null,
+  tableWidth: number | null,
+  widths: Array<number | null>[],
+  options: Options
+) => GeneralSteps.sequence([
+  sAssertTableStructure(editor, ApproxStructure.build((s, str) => s.element('table', {
+    attrs: { border: str.is('1') },
+    styles: { 'border-collapse': str.is('collapse') },
+    children: [
+      s.element('colgroup', {
+        children: Arr.range(cols, () =>
+          s.element('col', {})
+        )
+      }),
+      s.element('tbody', {
```

Comment:
This part appears to be a duplicate of the above, so could you take. a look at removing some of the duplication please?

I suspect this doesn't need to be a separate function and you could just use a flag with the current function instead. eg:
```
const colGroupStruct = useColumnGroup ? [
  s.element('colgroup', {
    children: Arr.range(cols, () => s.element('col', {}))
  })
] : [ ];
```